src/plugins/data/server/search/search_service.ts

```diff
@@ -328,36 +351,69 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
     return strategy.extend(id, keepAlive, options, deps);
   };
 
-  private getSearchStrategy = <
-    SearchStrategyRequest extends IKibanaSearchRequest = IEsSearchRequest,
-    SearchStrategyResponse extends IKibanaSearchResponse = IEsSearchResponse
-  >(
-    name: string = this.defaultSearchStrategyName
-  ): ISearchStrategy<SearchStrategyRequest, SearchStrategyResponse> => {
-    this.logger.debug(`Get strategy ${name}`);
-    const strategy = this.searchStrategies[name];
-    if (!strategy) {
-      throw new KbnServerError(`Search strategy ${name} not found`, 404);
+  private deleteSession = async (deps: SearchStrategyDependencies, sessionId: string) => {
```

Comment:
Since we don't really delete the saved object, maybe `cancelSession` is a better name?

Reply:
Right now we actually are deleting the session (see line 366). Should we only be cancelling instead? I guess instead of deleting we could just update the status to `CANCELLED`... Is that what you're thinking?

Reply:
Yes, of course. No need to delete the session object IMO.